circom-prover/src/lib.rs

```diff
@@ -34,3 +36,70 @@ impl CircomProver {
         prover::verify(proof_lib, zkey_path, proof, public_inputs)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    const ZKEY_PATH: &str = "./test-vectors/multiplier2_final.zkey";
+
+    fn generate_proof(witness_fn: WitnessFn, proof_lib: ProofLib) -> CircomProof {
+        let inputs = HashMap::from([
+            ("a".to_string(), vec!["1".to_string()]),
+            ("b".to_string(), vec!["2".to_string()]),
+        ]);
+        CircomProver::prove(proof_lib, witness_fn, inputs, ZKEY_PATH.to_string()).unwrap()
+    }
+
+    fn verify_proof(proof: Vec<u8>, public_inputs: Vec<u8>, proof_lib: ProofLib) -> bool {
+        CircomProver::verify(proof_lib, proof, public_inputs, ZKEY_PATH.to_string()).unwrap()
+    }
+
+    #[cfg(all(feature = "rustwitness", feature = "arkworks"))]
+    #[test]
+    fn test_rustwitness_arkworks_prove_and_verify() {
+        rust_witness::witness!(multiplier2);
+        let res = generate_proof(
+            WitnessFn::RustWitness(multiplier2_witness),
+            ProofLib::Arkworks,
+        );
+        let res = verify_proof(res.proof, res.pub_inputs, ProofLib::Arkworks);
+        assert!(res);
+    }
+
+    #[cfg(all(feature = "witnesscalc", feature = "arkworks"))]
+    #[test]
+    fn test_witnesscalc_arkworks_prove_and_verify() {
+        use num_bigint::BigInt;
+        witnesscalc_adapter::witness!(multiplier2);
+        // TODO: The conversion should be exported in witness.rs
```

Comment:
We can include the issue number here as well.

Reply:
[d51841a](https://github.com/zkmopro/mopro/pull/318/commits/d51841a5d61fefe0973e81c3195398664d61ca56)